tests/test_copy.py

```diff
@@ -0,0 +1,97 @@
+import os
+import sys
+import shutil
+import hashlib
+import unittest
+
+from kapitan.cli import main
+from kapitan.utils import directory_hash
+from kapitan.inputs.copy import Copy
+
+
+search_path = ""
+ref_controller = ""
+test_path = os.path.dirname(os.path.abspath(__file__)) + "/test_copy"
+compile_path = test_path + "/output"
+file_path = test_path + "/input"
+test_file_path = file_path + "/test_copy_input"
+test_file_compiled_path = compile_path + "/test_copy_input"
+test_file_content = """
+apiVersion: v1
+kind: Pod
+metadata:
+  name: alpine
+  namespace: default
+spec:
+  containers:
+  - image: alpine:3.2
+    command:
+      - /bin/sh
+      - "-c"
+      - "sleep 60m"
+    imagePullPolicy: IfNotPresent
+    name: alpine
+  restartPolicy: Always
+"""
+
+
+class CopyTest(unittest.TestCase):
+
+    def setUp(self):
+
+        try:
+            shutil.rmtree(test_path)
+        except FileNotFoundError:
+            pass
+
+        for folder in [file_path, compile_path]:
```

Comment:
The code starting from here up to the end of this function should probably be in a separate helper function and called from `test_copy_file_folder` and `test_copy_folder_folder`? 
`setUp` is usually minimal and then in each test you run the compilation.

Reply:
Separated the folder structure to a helper function.